use-shopping-cart/core/stripe-middleware.js

```diff
@@ -12,20 +12,42 @@ export const handleStripe = (store) => (next) => async (action) => {
     } catch (error) {
       console.log('error', error)
     }
-    if (action.payload?.sessionId) {
+    if (cart.cartMode === 'checkout-session') {
       return stripe.redirectToCheckout({
         sessionId: action.payload.sessionId
       })
     }
 
-    if (store.getState().mode === 'client-only') {
+    if (cart.cartMode === 'client-only') {
       const checkoutData = getCheckoutData(cart)
       stripe.redirectToCheckout(checkoutData)
     }
+
+    throw new Error(`Invalid value for "cartMode" was found: ${cart.cartMode}`)
```

Comment:
Added warning too to make them the same.

```suggestion
    console.warn('redirectToCheckout only works with checkout-session or client-only cartMode.', cart, action)
    throw new Error(`Invalid value for "cartMode" was found: ${cart.cartMode}`)
```